rueidislock/lock.go

```diff
@@ -304,28 +296,19 @@ func (m *locker) try(ctx context.Context, cancel context.CancelFunc, name string
 		if released := atomic.AddInt32(&released, 1); released >= m.majority {
 			cancel()
 			if released == m.totalcnt {
-				if atomic.LoadInt32(&failures) >= m.majority {
-					timer := time.NewTimer(m.interval)
-					cases := make([]reflect.SelectCase, 0, m.totalcnt+1)
-					cases = append(cases, reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(timer.C)})
-					for i := int32(0); i < m.totalcnt; i++ {
-						cases = append(cases, reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(g.csc[i])})
-					}
-					reflect.Select(cases)
```

Comment:
This is no longer needed because we already clear each of the`g.csc` channel before entering the `monitoring` function.